tests/snippets/dict.py

```diff
@@ -95,8 +95,15 @@
 x[2] = 2
 x[(5, 6)] = 5
 
-with assertRaises(KeyError):
-    x["not here"]
+assert_raises(KeyError,
+              lambda: x['not here'],
+              "'not here'")
+assert_raises(KeyError,
+              lambda: x[10],
+              "10")
+assert_raises(KeyError,
+              lambda: x[10.0],
+              "10.0")
```

Comment:
You should had some test that you actually return the `__repr__` of an object and not the `__str__`:
```py
class A: 
    def __repr__(self): 
        return "repr"

    def __str__(self): 
        return "str"

a = A()

try:
    x[a]
except KeyError as err:
    assert str(err) == "repr"
```